From b8d8bcf8563ab463765f7315af1901b9ef3423e9 Mon Sep 17 00:00:00 2001
From: Thomas Devoogdt <thomas.devoogdt@barco.com>
Date: Wed, 7 Sep 2022 02:32:44 -0700
Subject: [PATCH 11/25] Merge 254220@main - CoordinatedGraphics: fix includes
 when COORDINATED_GRAPHICS is not set

https://bugs.webkit.org/show_bug.cgi?id=244811

Reviewed by Fujii Hironori.

COORDINATED_GRAPHICS is not set if USE_OPENGL_OR_ES=OFF

- Forward declare CoordinatedGraphicsLayer, the header will be
  included in CompositingCoordinator if COORDINATED_GRAPHICS is set.

- The LayerTreeHost.cpp is not compiled if the COORDINATED_GRAPHICS
  flag is not set, so extend the mocked function list.

Signed-off-by: Thomas Devoogdt <thomas.devoogdt@barco.com>

Canonical link: https://commits.webkit.org/254220@main

(cherry picked from commit f06201ca85137bd3dbfdc52ccc475d18ed2332e4)
---
 .../WebPage/CoordinatedGraphics/LayerTreeHost.h          | 9 ++++++---
 1 file changed, 6 insertions(+), 3 deletions(-)

diff --git a/Source/WebKit/WebProcess/WebPage/CoordinatedGraphics/LayerTreeHost.h b/Source/WebKit/WebProcess/WebPage/CoordinatedGraphics/LayerTreeHost.h
index 293c2e5de26e..1dfae4a921b8 100644
--- a/Source/WebKit/WebProcess/WebPage/CoordinatedGraphics/LayerTreeHost.h
+++ b/Source/WebKit/WebProcess/WebPage/CoordinatedGraphics/LayerTreeHost.h
@@ -37,9 +37,6 @@
 #include <wtf/Forward.h>
 #include <wtf/OptionSet.h>
 #include <wtf/RunLoop.h>
-#if PLATFORM(GTK)
-#include <WebCore/CoordinatedGraphicsLayer.h>
-#endif
 
 #if USE(GRAPHICS_LAYER_TEXTURE_MAPPER)
 
@@ -48,6 +45,7 @@
 #else // USE(GRAPHICS_LAYER_TEXTURE_MAPPER)
 
 namespace WebCore {
+class CoordinatedGraphicsLayer;
 class IntRect;
 class IntSize;
 class GraphicsLayer;
@@ -234,6 +232,7 @@ inline void LayerTreeHost::scrollNonCompositedContents(const WebCore::IntRect&)
 inline void LayerTreeHost::forceRepaint() { }
 inline void LayerTreeHost::forceRepaintAsync(CompletionHandler<void()>&&) { }
 inline void LayerTreeHost::sizeDidChange(const WebCore::IntSize&) { }
+inline void LayerTreeHost::targetRefreshRateDidChange(unsigned) { }
 inline void LayerTreeHost::pauseRendering() { }
 inline void LayerTreeHost::resumeRendering() { }
 inline WebCore::GraphicsLayerFactory* LayerTreeHost::graphicsLayerFactory() { return nullptr; }
@@ -242,6 +241,10 @@ inline void LayerTreeHost::didChangeViewportAttributes(WebCore::ViewportAttribut
 inline void LayerTreeHost::setIsDiscardable(bool) { }
 inline void LayerTreeHost::deviceOrPageScaleFactorChanged() { }
 inline RefPtr<WebCore::DisplayRefreshMonitor> LayerTreeHost::createDisplayRefreshMonitor(WebCore::PlatformDisplayID) { return nullptr; }
+#if PLATFORM(GTK)
+inline void LayerTreeHost::adjustTransientZoom(double, WebCore::FloatPoint) { }
+inline void LayerTreeHost::commitTransientZoom(double, WebCore::FloatPoint) { }
+#endif
 #endif
 
 } // namespace WebKit
-- 
2.37.3

